mentorship ios/Service/Networking/RelationAPI.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  RelationAPI.swift
+//  Created on 23/07/20
+//  Created for AnitaB.org Mentorship-iOS 
+//
+
+import Foundation
+import Combine
+
+class RelationAPI {
+    private var cancellable: AnyCancellable?
+    private var tasksCancellable: AnyCancellable?
+    
+    func fetchCurrentRelation(completion: @escaping (RequestStructure) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        //api call
+        cancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.currentRelation, token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(RelationModel().currentRelation) }
+            .sink {
+                completion($0)
+        }
+    }
+    
+    func fetchTasks(id: Int, completion: @escaping ([TaskStructure], Bool) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
```

Comment:
Shell we return here a closure with fail?

Reply:
We can. How do you feel about using a fatal error here instead?

Reply:
would UI be able to catch it?

Reply:
A fatal error terminates the app.

Reply:
In our current code, we have used fatalError() in our logout function (if delete token fails).